src/source/JavaGenerator.scala

```diff
@@ -97,7 +97,10 @@ class JavaGenerator(spec: Spec) extends Generator(spec) {
       writeDoc(w, c.doc)
       javaAnnotationHeader.foreach(w.wl)
       marshal.nullityAnnotation(c.ty).foreach(w.wl)
-      w.w(s"public static final ${marshal.fieldType(c.ty)} ${idJava.const(c.ident)} = ")
+
+      // If the constants are part of a Java interface, omit the "static" and "final" specifiers.
+      val staticFinalString = if (forJavaInterface) "" else "static final "
+      w.w(s"public ${staticFinalString}${marshal.fieldType(c.ty)} ${idJava.const(c.ident)} = ")
```

Comment:
Andrew suggested omitting "static" and "final" to avoid warnings, and Android Studio does give me warnings when I look at an interface that includes "static," "final," or "public" in declarations of constants.